src/VisualStudio/LiveShare/Impl/Client/Razor/CSharpLspRazorProjectFactory.cs

```diff
@@ -0,0 +1,39 @@
+﻿// Copyright (c) Microsoft.  All Rights Reserved.  Licensed under the Apache License, Version 2.0.  See License.txt in the project root for license information.
+
+using System;
+using System.Collections.Generic;
+using System.ComponentModel.Composition;
+using Microsoft.CodeAnalysis;
+using Microsoft.VisualStudio.Shell;
+
+namespace Microsoft.VisualStudio.LanguageServices.LiveShare.Client.Razor
+{
+    [Export]
+    internal class CSharpLspRazorProjectFactory
+    {
+        private readonly RemoteLanguageServiceWorkspaceHost _remoteLanguageServiceWorkspaceHost;
+        private readonly Dictionary<string, ProjectId> _projects = new Dictionary<string, ProjectId>();
+
+        public ProjectId GetProject(string projectName)
+        {
+            if (_projects.TryGetValue(projectName, out var projectId))
```

Comment:
```_projects``` appears to be accessed, but never added to, so it seems like we'll always create a new one?

Reply:
D'oh.